packages/valory/skills/abstract_round_abci/models.py

```diff
@@ -532,6 +541,15 @@ class TendermintRecoveryParams(TypeCheckMixin):
     reset_params: Optional[List[Tuple[str, str]]] = None
     serialized_db_state: Optional[str] = None
 
+    def __hash__(self) -> int:
+        """Hash the object."""
+        return hash(
+            self.reset_from_round
+            + str(self.round_count)
+            + str(self.serialized_db_state)
+            + json.dumps(self.reset_params)
```

Comment:
Sort keys?

Reply:
Not necessary for determinism since we are using python >= 3.7

Reply:
This is a list of tuples actually, so that shouldnt make a difference.

Reply:
It's recursive afaik and we should always use it imo. Technically, there could be a non-Python client interacting. This just makes it all more transparent.

Reply:
After merging `fix/api-specs` the `reset_params` is actually `Dict[str, str]`, `sort_keys=True` was added.